hyperspy/drawing/utils.py

```diff
@@ -343,6 +343,466 @@ def _set_spectrum_xlabel(spectrum, ax):
+                    basename = basename[:-1]
+                elif basename[len(basename) - 1] == ' ':
+                    basename = basename[:-1]
+                else:
+                    break
+
+        # namefrac is ratio of length of basename to the image name
+        # if it is high (e.g. over 0.5), we can assume that all images
+        # share the same base
+        namefrac = float(len(basename)) / len(label_list[0])
+
+        if namefrac > 0.5:
+            # there was a significant overlap of label beginnings
+            shared_titles = True
+            # only use new suptitle if one isn't specified already
+            if suptitle is None:
+                suptitle = basename
+
+        else:
+            # there was not much overlap, so default back to 'titles' mode
+            shared_titles = False
+            label = 'titles'
+            div_num = 0
+
+    elif label is 'titles':
+        # Set label_list to each image's pre-defined title
+        label_list = [x.metadata.General.title for x in images]
+
+    elif isinstance(label, str):
+        # Set label_list to an indexed list, based off of label
+        label_list = [label + " " + repr(num) for num in range(n)]
+
+    elif isinstance(label, list) and all(isinstance(x, str) for x in label):
+        label_list = label
+        user_labels = True
+        # If list of labels is longer than the number of images, just use the
+        # first n elements
+        if len(label_list) > n:
+            del label_list[n:]
+        if len(label_list) < n:
+            label_list *= (n / len(label_list)) + 1
+            del label_list[n:]
+
+    else:
+        # catch all others to revert to default if bad input
+        print "Did not understand input of labels. Defaulting to image titles."
+        label_list = [x.metadata.General.title for x in images]
+
+    # Determine appropriate number of images per row
+    rows = int(np.ceil(n / float(per_row)))
+    if n < per_row:
+        per_row = n
+
+    # Set overall figure size and define figure (if not pre-existing)
+    if fig is None:
+        k = max(plt.rcParams['figure.figsize']) / max(per_row, rows)
+        f = plt.figure(figsize=(tuple(k * i for i in (per_row, rows))))
+    else:
+        f = fig
+
+    # Initialize list to hold subplot axes
+    axes_list = []
+
+    # Initialize list of rgb tags
+    isrgb = [False] * len(images)
+
+    # Check to see if there are any rgb images in list
+    # and tag them using the isrgb list
+    for i, img in enumerate(images):
+        if rgb_tools.is_rgbx(img.data):
+            isrgb[i] = True
+
+    # Find global min and max values of all the non-rgb images for use with 'single' scalebar
+    if colorbar is 'single':
+        gl_max, gl_min = max([i.data.max() for i in list(compress(images, [not j for j in isrgb]))]), \
+                         min([i.data.min() for i in list(compress(images, [not j for j in isrgb]))])
+
```

Comment:
Are you aware of the way `plot()` set the vmin and vmax with `perc` arguments? It would be nice to have a similar method.